[Jahnavi Pydi] iP

pjahn31 opened the pull request.

pjahn31 committed:
Duke renamed to Emily_duke_ref

pjahn31 committed:
Level 0- Emily is made.

pjahn31 committed:
Level 1- Emily echoes

pjahn31 committed:
Level 2- Emily can add and produce a list

pjahn31 committed:
Level 3- Emily can mark tasks as done

pjahn31 committed:
Level 3- Task class

lakshm1i commented on src/main/java/Emily.java:
```diff
@@ -0,0 +1,61 @@
+import java.util.ArrayList;
+
+public class Emily {
+    public static ArrayList<Task> tasks = new ArrayList<>();
```
I noticed you're using ArrayList directly here. Would using the interface type List<Task> instead of ArrayList<Task> make it easier to switch to another list implementation in the future?

lakshm1i commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,21 @@
+public class Task {
+    protected String description;
+    protected Boolean isDone;
```
Would it be better to use the primitive type boolean instead of Boolean? It avoids unnecessary object creation and aligns with Java conventions for simple true/false flags.

Markneoneo replied:
Using boolean is better than Boolean in Java because boolean is a primitive type, while Boolean is an object wrapper class.

lakshm1i commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,21 @@
+public class Task {
+    protected String description;
```
Should the description and isDone fields be private instead of protected to fully encapsulate the data? It might prevent unintended modifications from outside the class.

lakshm1i commented on src/main/java/Emily.java:
```diff
@@ -0,0 +1,61 @@
+            Task task = tasks.get(taskNumber - 1);
+            task.unmarkAsDone();
+            System.out.println("Ok, I've unmarked this task as not done yet:");
+            System.out.println("  [ ] " + task.description);
```
Would it be better to access description via a getter method instead of direct field access for consistency with encapsulation principles?

lakshm1i commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,21 @@
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); //mark done task with X
```
I like how concise this is!

lakshm1i reviewed (commented):
Great job!

adoorknob commented on src/main/java/Emily.java:
```diff
@@ -0,0 +1,61 @@
+
+    public static void main(String[] args) {
+        Scanner sc = new Scanner(System.in);
+        System.out.println("Hello! I'm Emily.");
```
maybe you could abstract this out into a 'welcome message' method for readability!

adoorknob commented on src/main/java/Emily.java:
```diff
@@ -0,0 +1,61 @@
+        while(true) {
+            String input = sc.nextLine();
+
+            if (input.equalsIgnoreCase("bye")) {
+                System.out.println("Bye. Hope to see you again soon!");
+                break;
+            } else if (input.equalsIgnoreCase("list")) {
+                printTasks();
+            } else if (input.startsWith("mark ")) {
+                markTaskAsDone(Integer.parseInt(input.substring(5)));
+            } else if (input.startsWith("unmark ")) {
+                unmarkTask(Integer.parseInt(input.substring(7)));
+            } else {
+                addTask(input);
+            }
```
maybe you could use a switch statement instead, for better scalability!
for example,
```
command = input.toLowercase();
switch (command) {
  case "xxx": // ...
  case "yyy": // ...
}
```

adoorknob commented on src/main/java/Emily.java:
```diff
@@ -0,0 +1,61 @@
+        }
+    }
+
+    public static void markTaskAsDone(int taskNumber) {
```
the name of this method is a bit different from the `unmarkTask` function, which might be a bit confusing

adoorknob commented on src/main/java/Emily.java:
```diff
@@ -0,0 +1,61 @@
+    public static void unmarkTask(int taskNumber) {
+        if (taskNumber > 0 && taskNumber <= tasks.size()) {
+            Task task = tasks.get(taskNumber - 1);
+            task.unmarkAsDone();
```
i like the abstraction of this into another method, really upholding the OOP spirit

adoorknob commented on src/main/java/Emily.java:
```diff
@@ -0,0 +1,61 @@
+    }
+
+    public static void unmarkTask(int taskNumber) {
+        if (taskNumber > 0 && taskNumber <= tasks.size()) {
```
nice catch for the task number! maybe consider making this throw an exception instead, so the user knows why the command didn't work

adoorknob reviewed (commented):
lgtm apart from some nits, i like your naming and the readability of your code!

Markneoneo commented on src/main/java/Emily.java:
```diff
@@ -2,42 +2,60 @@
+                addTask(input);
             }
         }
-        sc.close();
```
Maybe use switch cases instead of many if else statements. The code will be clearer and more structured.

Markneoneo commented on src/main/java/Emily.java:
```diff
@@ -2,42 +2,60 @@
-        tasks.add(task);
-        System.out.println("added: " + task);
+    public static void markTaskAsDone(int taskNumber) {
+        if (taskNumber > 0 && taskNumber <= tasks.size()) {
```
The code checks if taskNumber is within range but does not handle invalid inputs.
Fix: Add an else block to notify the user when an invalid task number is provided.
        System.out.println("⚠ Error: Invalid task number.");

Markneoneo commented on src/main/java/Emily.java:
```diff
@@ -2,42 +2,60 @@
+        }
+    }
+
+    public static void unmarkTask(int taskNumber) {
```
Method Naming Consistency:
markTaskAsDone and unmarkTask have inconsistent naming styles

Markneoneo commented on src/main/java/Emily.java:
```diff
@@ -2,42 +2,60 @@
 
 public class Emily {
-    public static ArrayList<String> tasks = new ArrayList<>();
+    public static ArrayList<Task> tasks = new ArrayList<>();
```
Nice implementation of the new Class Task.

Markneoneo reviewed (commented):
The code has syntax issues (missing closing brace, inconsistent indentation) and readability concerns (redundant print statements, unclear indexing). It lacks proper error handling for invalid task numbers and may risk a NullPointerException if tasks is uninitialized. Naming conventions are inconsistent, and duplicate logic can be extracted into helper methods for maintainability.

pjahn31 committed:
Changes made after Week_4 PR

Will make more changes after Level-4 during A-CodeQuality

pjahn31 committed:
Added features of Todo, Deadline and Event

Subclass of Task

pjahn31 committed:
Changes made to Task class and Emily class

To fit the addition of subclass Todo, Deadline, Event

Alfred-Goh02 commented on src/main/java/Emily.java:
```diff
@@ -0,0 +1,69 @@
+            }
+        }
+    }
+
```
Consider moving these methods to another class, and separate print lines to another class as well. This is for easier management and to ensure that each file is representative of what methods it contains.

Alfred-Goh02 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,14 @@
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by + ")";
```
Avoid using magic strings and assign it to a meaningful constant.

Alfred-Goh02 commented on src/main/java/Emily_duke_ref.java:
```diff
@@ -1,4 +1,4 @@
-public class Duke {
```
If this file is used just for your reference, you don't have to commit it.

Alfred-Goh02 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,16 @@
+
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + "(from: " + from + " to: " + to + ")";
```
Magic Strings here as well.

Alfred-Goh02 commented on src/main/java/Todo.java:
```diff
@@ -0,0 +1,10 @@
+    public Todo(String description) {
+        super(description);
+    }
+
```
The method is compact and easy to understand, nice!

Alfred-Goh02 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,26 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
```
Very concise methods which follow conventions, nice!

pjahn31 committed:
Tweaked the code to comply with coding standards

pjahn31 committed:
Improved code quality

Used switch statements instead of if-else statements
Tried to make functions out of simple print lines

pjahn31 committed:
Added error handling to deadline and event

<the more complicated functions>

pjahn31 committed:
Made seperate functions for adding of todo, deadline and event
Todo now handles errors

pjahn31 committed:
Merge branch 'branch-Level-5'

pjahn31 committed:
Used exceptions to handle errors

<Made the code neater>

pjahn31 committed:
Added support for deleting tasks from the list

pjahn31 committed:
Using Java Collections

already has been used, added a spacing to commit this for the ip

pjahn31 committed:
Added local saving of list in the hard disk

pjahn31 committed:
Merge branch 'branch-Level-5'

pjahn31 committed:
Merge branch 'branch-Level-6'

pjahn31 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Emily.java

pjahn31 committed:
Made my code more OOP
Added packages and refactored the code to extract out closely related code as classes

pjahn31 committed:
Added a find command

pjahn31 committed:
Revert "Added a find command"

This reverts commit 3c382097a0653da1db80e07240a193aee0688941.

pjahn31 committed:
Added a find command
Users can type find to get a list of tasks with the corresponding keyword

pjahn31 committed:
Merge pull request #1 from pjahn31/branch-Level-9

PR from branch-Level-9 to master

pjahn31 committed:
Merge branch 'master' of https://github.com/pjahn31/ip

pjahn31 committed:
Added JavaDoc comments to the code

pjahn31 committed:
JavaDoc comments

Forgot to create a seperate commit for the branch merge, so will be using this commit instead

pjahn31 committed:
Merge branch 'branch-A-JavaDoc'

pjahn31 committed:
Updated User Guide of Emily Task Manager

pjahn31 committed:
Updated User Guide

pjahn31 committed:
Update README.md
